lib/org/chromium/webidl/URLExtractor.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'URLExtractor',
+
+  requires: ['org.chromium.webidl.IDLFileContents'],
+
+  properties: [
+    {
+      name: 'excludeRegexp',
+      // documentation: 'Excludes all URLs matching this expression.',
+    },
+    {
+      name: 'includeRegexp',
+      // documentation: 'Only include URLs matching this expression.',
+    }
+  ],
+
+  constants: {
+    URL_REGEX: /https?:\/\/(www\.)?[-_a-zA-Z0-9.]{2,256}\.[a-z]{2,6}(\/[^?#, \n]*)?(\?[^#, \n]*)?/g,
```

Comment:
I believe it just needs to be sufficient to extract well-formed urls from IDL files. The previous regex was `https?:\/\/[^/]+(\/[^?#, \n]*)?(\?[^#, \n]*)?` but this was even less restrictive and resulted in matches of weird results (thus the change in this revision)

I can attempt at writing a stricter regex (maybe `https?:\/\/(www\.)?([a-zA-Z0-9][-_a-zA-Z0-9])([-_a-zA-Z0-9.])*\.[a-z]{2,6}(\/[^?#, \n]*)?(\?[^#, \n]*)?` would be a better option?)